app/javascript/components/patient/vaccines/VaccineModal.js

```diff
@@ -0,0 +1,244 @@
+                />
+              </Form.Group>
+            </Row>
+            <Row>
+              <Form.Group as={Col}>
+                <Form.Label htmlFor="product-name-select" className="nav-input-label">
+                  Product Name
+                </Form.Label>
+                <Select
+                  inputId="product-name-select"
+                  name="product_name"
+                  defaultValue={
+                    this.props.currentVaccineData?.product_name
+                      ? { label: this.props.currentVaccineData.product_name, value: this.props.currentVaccineData.product_name }
+                      : ''
+                  }
+                  value={this.state.product_name ? { label: this.state.product_name, value: this.state.product_name } : ''}
+                  options={this.getProductNameOptions().map(option => {
+                    return { label: option, value: option };
+                  })}
+                  onChange={this.handleProductNameChange}
+                  placeholder=""
+                  theme={theme => ({
+                    ...theme,
+                    borderRadius: 0,
+                  })}
+                  styles={{ menu: base => ({ ...base, zIndex: 9999 }), option: base => ({ ...base, minHeight: 30 }) }}
+                />
+              </Form.Group>
+            </Row>
+            <Row>
+              <Form.Group as={Col}>
+                <Form.Label htmlFor="administration-date" className="nav-input-label">
+                  Administration Date
+                </Form.Label>
+                <DateInput
+                  id="administration-date"
+                  date={this.state.administration_date}
+                  minDate={'1900-01-01'}
+                  maxDate={moment()
+                    .add(30, 'days')
+                    .format('YYYY-MM-DD')}
+                  onChange={this.handleAdministrationDateChange}
+                  placement="bottom"
+                  customClass="form-control-lg"
+                  ariaLabel="Administration Date Input"
+                />
+              </Form.Group>
+            </Row>
+            <Row>
+              <Form.Group as={Col}>
+                <Form.Label htmlFor="dose-number-select" className="nav-input-label">
+                  Dose Number
+                </Form.Label>
+                <Select
+                  inputId="dose-number-select"
+                  name="dose_number"
+                  defaultValue={
+                    this.props.currentVaccineData?.dose_number
+                      ? { label: this.props.currentVaccineData.dose_number, value: this.props.currentVaccineData.dose_number }
+                      : ''
+                  }
+                  value={this.state.dose_number ? { label: this.state.dose_number, value: this.state.dose_number } : ''}
+                  options={this.props.dose_number_options?.map(option => {
```

Comment:
would it make sense to do some kind of validation with dose number in that dose #2 for example cannot come before dose #1 based on the administration date?

Reply:
This was also discussed in part of the requirement gathering meetings we had, and we decided against it for now.